contracts/remarkables/src/msg.rs

```diff
@@ -0,0 +1,241 @@
+use crate::ContractError;
+use cosmwasm_std::{Addr, Coin, Uint64};
+use cw721_base::InstantiateMsg as Cw721InstantiateMsg;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InstantiateMsg {
+    /// Address of who will have the right to administer the contract.
+    pub admin: String,
+    /// Id of the CW721 contract to initialize together with this contract.
+    pub cw721_code_id: Uint64,
+    /// Initialization message that will be sent to the CW721 contract.
+    pub cw721_instantiate_msg: Cw721InstantiateMsg,
+    /// Id of the subspace to operate.
+    pub subspace_id: Uint64,
+    /// List of rarities to initialize with this contract.
+    pub rarities: Vec<Rarity>,
+}
+
+impl InstantiateMsg {
+    /// Checks that the data inside the message are coherent.
+    /// NOTE: This function don't checks if the address are valid.
+    pub fn validate(&self) -> Result<(), ContractError> {
+        if self.subspace_id.is_zero() {
```

Comment:
Can youa also do the same check for `cw721_code_id` since is > 0

Reply:
Added